python/sdss_access/path/path.py

```diff
@@ -1529,34 +1557,200 @@ def pad_fieldid(self, filetype, **kwargs):
+                                     'spallfield_coadd','spalllinefield_coadd']:
+                return(coaddname)
+            else:
+                return('fields')
 
+
+    def spcoaddgrp(self, filetype, **kwargs):
+        ''' Returns the coadd group (field group analog) subfolder structure for the BOSS idlspec2d run2d version
+
+        Parameters
+        ---------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        coadd : str
+            Name of the custom coadd schema
         Returns
         -------
-        tileidgrp : str
-            Tile ID group directory in the format ``NNNNXX``.
+        spcoaddgrp : str
+        '''
+        run2d = kwargs.get('run2d', None)
+        coaddname = kwargs.get('coadd', None)
+
+        if (not run2d):
+            return ''
+        elif (('v5' in run2d) or (str(run2d) in ['26','103','104']) or
+              ('v6_0' in run2d) or ('v6_1' in run2d)):
+            return ''
+        else:
+            return(coaddname)
+            
+    def sptypefolder(self, filetype, **kwargs):
+        ''' Returns the reorganized subfolder structure for the BOSS idlspec2d run2d version
 
+        Parameters
+        ---------
+        filetype : str
+            File type parameter
+        run2d : str
+            BOSS idlspec2d run2d version
+        Returns
+        -------
+        sptypefolder : str
         '''
 
-        tileid = kwargs.get('tileid', None)
-        if not tileid:
-            return '0000XX'
-        elif '*' in str(tileid):
-            return '{0}XX'.format(tileid)
-        return '{:0>4d}XX'.format(int(tileid) // 1000)
+        run2d = kwargs.get('run2d', None)
+        
+        if (not run2d):
+            return ''
+        elif ('v5' in run2d) or (str(run2d) in ['26','103','104']):
+            return ''
+        elif ('v6_0' in run2d) or ('v6_1' in run2d):
+            if filetype.lower() in ['speclite_epoch','specfull_epoch',
+                                    'spallfield_epoch','spalllinefield_epoch']:
+                return('epoch/spectra')
+            else:
+                return ''
+        else:
+            if filetype.lower() in ['fieldlist_epoch','spall_epoch',
+                                    'spall-lite_epoch','spallline_epoch']:
+                return('summary/epoch')
+            elif filetype.lower() in ['speclite_epoch','specfull_epoch',
+                                      'spallfield_epoch','spalllinefield_epoch']:
+                return('spectra/epoch')
+            elif filetype.lower() in ['conflist','fieldlist','spall',
+                                      'spall-lite','spallline']:
+                return('summary/daily')
+            elif filetype.lower() in ['speclite','specfull',
+                                     'spallfield','spalllinefield']:
+                return('daily')
+            else:
+                return('fields')
```

Comment:
I'll add the comment here but this applies to all the other functions.  Since the `return` is within the condition, you actually don't need the `elif` and `else` conditions. Since it returns immediately, the rest of the code doesn't get checked.  This can help simplify the code and make it more readable.   Everything after the `else` here can be brought back up one level.

Reply:
Fixed this and other similar instances.